### PR TITLE
Adjust code generation to be similar to old CSE.

### PR DESCRIPTION
**Pull request by kffaxen**

### CODE REVIEW

File: src/Feldspar/Core/Interpretation.hs [hunk 1]
```diff
@@ -40,6 +40,33 @@
 
 -- | Defines different interpretations of Feldspar programs
 
+{-# LANGUAGE CPP #-}
+
+#if 0
```
Comment (pjonsson) [inline src/Feldspar/Core/Interpretation.hs]: Should be handled by the build system.

Comment (kffaxen) [reply src/Feldspar/Core/Interpretation.hs]: Temporary solution. Will be removed in the next commit.

File: src/Feldspar/Core/Interpretation.hs [hunk 2]
```diff
@@ -40,6 +40,33 @@
...
+#define INCREMENTAL_CSE
+#endif
+
+#ifdef INCREMENTAL_CSE
+module Feldspar.Core.Interpretation where
+
+-- | Possible compilation targets in a broad sense.
```
Comment (pjonsson) [inline src/Feldspar/Core/Interpretation.hs]: Cumulative and some other things shouldn't conflict with the incremental CSE, put the ifdefs in the export list and around the existing definitions instead.

File: src/Feldspar/Core/Representation.hs [hunk 3]
```diff
@@ -546,9 +564,20 @@ shOp _ = True
...
+-- of magnitudes cheaper than the array copy generated for the let-binding.
+-- With a better compilation of array assignments, the need for this
+-- special case goes away.
+goodToShare (_ :& Operator op :@ e :: AExpr a)
+  | isArrayT (typeRepF :: TypeRep a) && isSelOp op && not (goodToShare e)
+  = False
 goodToShare (_ :& _ :@ _) = True
 goodToShare _                   = False
 
+isArrayT (ArrayType _) = True
```
Comment (pjonsson) [inline src/Feldspar/Core/Representation.hs]: Missing type signature?

Comment (kffaxen) [reply src/Feldspar/Core/Representation.hs]: You're right. Will fix!

**Pull request merged**